# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../isakmpd/compile.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
--- isakmpd-20041012.orig/sysdep/common/libsysdep/arc4random.c	2004-10-12 23:25:52.000000000 +0200
+++ isakmpd-20041012.orig-fixed/sysdep/common/libsysdep/arc4random.c	2004-10-19 10:35:28.681835096 +0200
@@ -78,7 +78,7 @@
 static void
 arc4_stir(struct arc4_stream *as)
 {
-	int     fd;
+	int     fd, i;
 	struct {
 		struct timeval tv;
 		u_int8_t rnd[128 - sizeof(struct timeval)];
--- isakmpd-20041012.orig/x509.c.vanilla	2005-10-31 11:38:52.000000000 +0100
+++ isakmpd-20041012.orig/x509.c	2005-10-31 11:39:08.000000000 +0100
@@ -910,7 +910,7 @@
 	X509_STORE_CTX_init(&csc, x509_cas, cert, NULL);
 #if OPENSSL_VERSION_NUMBER >= 0x00907000L
 	/* XXX See comment in x509_read_crls_from_dir.  */
-	if (x509_cas->flags & X509_V_FLAG_CRL_CHECK) {
+	if (x509_cas->param->flags & X509_V_FLAG_CRL_CHECK) {
 		X509_STORE_CTX_set_flags(&csc, X509_V_FLAG_CRL_CHECK);
 		X509_STORE_CTX_set_flags(&csc, X509_V_FLAG_CRL_CHECK_ALL);
 	}
--- isakmpd-20041012.orig/GNUmakefile.vanilla	2005-10-31 11:40:57.000000000 +0100
+++ isakmpd-20041012.orig/GNUmakefile	2005-10-31 11:40:59.000000000 +0100
@@ -56,7 +56,7 @@
 BINDIR=		/sbin
 endif
 ifndef LDSTATIC
-LDSTATIC=	-static
+LDSTATIC=	
 endif
 
 SRCS=		app.c attribute.c cert.c connection.c \
